lib/date.nit

```diff
@@ -0,0 +1,163 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Date manipulation
+module date
+
+# The main class Date (comming soon...)
+#
+# 	var d = new Date(2014, 01, 09)
+# 	assert d.to_s == "2014/01/09"
+#
+class Date
+	super Comparable
+
+	# Years as an integer (Int) ex: 1989
+	var year: Int
+
+	# Months as an integer (Int) ex: "3" for March
+	var month: Int
+
+	# Days as an integer (Int)
+	var days: Int
+
+	# Hours as an integer (Int)
+	var hours: Int = 0
+	
+	# Minutes as an integer (Int)
+	var mint: Int = 0
+
+	# Secound as an integer (Int)
+	var sec: Int = 0
+
+	# UTC time zone
+	var t_zone = "Z"
+
+	# for set the time (hours, min, sec) 
+	init time(y, mth, d, h, m, s: Int) do
+		year = y
+		month = mth
+		days = d
+		hours = h
+		mint = m
+		sec = s
+	end
+
+	# this will initialized the current date and time
+	init today do
+		var tm = new Tm.localtime
+		year = 1900 + tm.year
+		month = tm.mon + 1
+		days = tm.mday
+		hours = tm.hour
+		mint = tm.min
+		sec = tm.sec
+	end
+
+	# date and time expressed according to ISO 8601
+	redef fun to_s 
+	do
+		var y = year.to_s
+		var mth = month.to_s
+		var d = days.to_s
+		var h = hours.to_s
+		var m = mint.to_s
+		var s = sec.to_s
```

Comment:
Since we're dealing with a String here, the variables don't have to be `to_s`'d and could be used in the superstring directly.

This would allow the inlining of the method at the same time !